src/core/utils/intersectUtils.js

```diff
@@ -67,14 +67,12 @@ export function intersectRay( nodeIndex32, array, ray ) {
 
 	if ( ( tmin > tzmax ) || ( tzmin > tmax ) ) return false;
 
-	// if ( tzmin > tmin || tmin !== tmin ) tmin = tzmin; // Uncomment this line if add the distance check
+	if ( tzmin > tmin || tmin !== tmin ) tmin = tzmin;
 
 	if ( tzmax < tmax || tmax !== tmax ) tmax = tzmax;
 
 	//return point closest to the ray (positive side)
 
-	if ( tmax < 0 ) return false;
-
-	return true;
+	return tmin <= far && tmax >= near;
```

Comment:
Does this work correctly if the ray origin is within the box? I'm having trouble visualizing the logic here. It may be easiest to understand if we just return a distance here (`null` if no hit) and check the distance against near / far in the calling function.

Reply:
Though I see this makes the other code a bit more messy. Maybe this is easier to understand:

```js
const hitDist = tmin >= 0 ? tmin : tmax;
return hitDist > near && hitDist < far;
```

Reply:
I think this would be a good change to make to make thins more clearly with three.js' version that returns a `this.at( tmin >= 0 ? tmin : tmax, target )` as the hit point